lib/inferred_crumpets/builder.rb

```diff
@@ -71,44 +72,40 @@ def build_crumb_for_subject!
 
     def url_for_subject
       return unless can_route?(:show, id: subject.id) && linkable?
-      view_context.url_for(shallow? ? transformed_subject : subject_with_parents)
+      view_context.url_for(transformed_subject)
     end
 
     def url_for_collection
       return view_context.objects_path if view_context.objects_path.present?
       return unless can_route?(:index)
-      view_context.url_for(shallow? ? transformed_subject.class : class_with_parents)
+      view_context.url_for(linkable? ? transformed_subject.class : class_with_parents)
```

Comment:
This has a bogus line at 81 and can probably have the conditions written better.